components/nimbus/src/evaluator.rs

```diff
@@ -48,6 +55,7 @@ pub fn evaluate_enrollment(
     available_randomization_units: &AvailableRandomizationUnits,
     app_context: &AppContext,
     exp: &Experiment,
+    previous_enrollment: Option<&EnrollmentStatus>,
```

Comment:
I think if we construct the `TargetingAttributes` in this method, we'll be forever adding new arguments to this method.

If we change the `app_context: &AppContext` to `TargetingAttributes`, and create the `TargetingAttibutes` one level up, say in `evolve_enrollments` we should be able to add things to `TargetingAttributes` more easily (e.g. `update_age`, `install_age`, `previous_experiments`).

Reply:
Good call! Moved up to `evaluate_enrollment`!